rust/src/store.rs

```diff
@@ -204,22 +210,13 @@ impl BlockStore for IndexerStore {
         }
         self.database.put_cf(&blocks_cf, key, value)?;
 
-        fn global_slot_block_key(block: &PrecomputedBlock) -> Vec<u8> {
-            let global_slot = block.global_slot_since_genesis();
-            let mut bytes = global_slot_prefix(global_slot);
-            let mut state_hash_bytes = block.state_hash.as_bytes().to_vec();
-            bytes.append(&mut state_hash_bytes);
-            bytes
-        }
-
         // add to global slots block index
-        {
-            let key = global_slot_block_key(block);
-            let value = block.state_hash.as_bytes();
-            let blocks_global_slot_idx_cf = self.blocks_global_slot_idx_cf();
-            self.database
-                .put_cf(&blocks_global_slot_idx_cf, key, value)?;
-        }
+        self.database.put_cf(
+            self.blocks_global_slot_idx_cf(),
+            global_slot_block_key(block),
+            b"",
```

Comment:
This is likely a bug. We should be inserting the `state_hash`

Reply:
This is a feature, we don't need to store the state hash in both the key and the value. The key has the form: `{4 global slot BE bytes}{state hash bytes}`.